Implement tomcat and apache webserver media

jeddcuison opened the pull request:
Changes include:
- have the ability to define Tomcat media for JVM
- have the ability to define Apache HTTPD media for web server
- Media types should show "display name" instead of name 
- Cleanup properties
- Refactor Media to accurately reflect JpaMedia
- Use model mapper to map Media and JpaMedia vice versa
- update the tests due to code changes
- Do necessary UI modifications for the above mentioned changes
- Write a generic distributor method (used by JDK and Web Server)
- MediaType serializer
- Some refactoring (e.g move some stuff, code cleanup etc...)

jeddcuison committed:
Write a generic method that can distribute any media

jeddcuison committed:
Write a generic method that can distribute any media

jeddcuison committed:
List all media

jeddcuison committed:
List all media

jeddcuison committed:
Add "user friendly" name property

jeddcuison committed:
Add "user friendly" name property

jeddcuison committed:
Incorporate Tomcat media to Jvm and refactor media related code

In addition define Apache HTTPD media as part of web server

Code Change Details

- Use ModelMapper to handle DTO-entity mapping
- Refactor Media DTO to mirror JpaMedia class variable types
- Mark distributeWebServer as deprecated
- Transfer path serializer and deserializer to commons
- Update tests affected by the changes
- Refactor UI, rename tomcat variable names and define React keys to options element
- Format some code
- Remove unused import(s)
- Update java doc
- Remove equals and hashCode overrides in Media

jeddcuison committed:
Incorporate Tomcat media to Jvm and refactor media related code

In addition define Apache HTTPD media as part of web server

Code Change Details

- Use ModelMapper to handle DTO-entity mapping
- Refactor Media DTO to mirror JpaMedia class variable types
- Mark distributeWebServer as deprecated
- Transfer path serializer and deserializer to commons
- Update tests affected by the changes
- Refactor UI, rename tomcat variable names and define React keys to options element
- Format some code
- Remove unused import(s)
- Update java doc
- Remove equals and hashCode overrides in Media

jeddcuison committed:
Make Tomcat required and fix bug wherein JDK is  not validated

In addition rewrite comparators to set selected media from "==" to "==="

jeddcuison committed:
Make Tomcat required and fix bug wherein JDK is  not validated

In addition rewrite comparators to set selected media from "==" to "==="

jeddcuison committed:
Fix media entity to DTO mapping problems

The following is not being mapped

- MediaType
- localPath
- remoteDir

jeddcuison committed:
Fix media entity to DTO mapping problems

The following is not being mapped

- MediaType
- localPath
- remoteDir

jmahmoodcc committed:
when deploying the JVM use the tomcat media configured for the JVM

jmahmoodcc committed:
when deploying the JVM use the tomcat media configured for the JVM

jmahmoodcc committed:
move MediaType to common

jmahmoodcc committed:
move MediaType to common

jmahmoodcc committed:
Merge branch 'implement-tomcat-and-apache-webserver-media' of github.com:cerner/jwala into implement-tomcat-and-apache-webserver-media

jeddcuison committed:
Implement apache httpd media for web server

Details

- UI drop down for apache httpd
- Creation of new web server with apache httpd
- Update of new web server with apache httpd
- Refactor: write new create and update web server methods that returns the web server entity

jeddcuison committed:
Implement apache httpd media for web server

Details

- UI drop down for apache httpd
- Creation of new web server with apache httpd
- Update of new web server with apache httpd
- Refactor: write new create and update web server methods that returns the web server entity

jmahmoodcc committed:
remove remote.tomcat.dir.name from vars.properties

jmahmoodcc committed:
remove remote.tomcat.dir.name from vars.properties

jmahmoodcc committed:
remove jwala.tomcat.zip.name from vars.properties

jmahmoodcc committed:
remove jwala.tomcat.zip.name from vars.properties

jmahmoodcc committed:
remote remote.tomcat.service.exe from vars.properties

jmahmoodcc committed:
remote remote.tomcat.service.exe from vars.properties

jmahmoodcc committed:
fix install jvm script to strip dash from apache tomcat version

jmahmoodcc committed:
fix install jvm script to strip dash from apache tomcat version

jmahmoodcc committed:
allow nullable apacheHttpMediaId when creating a web server from the REST API

jmahmoodcc committed:
allow nullable apacheHttpMediaId when creating a web server from the REST API

jmahmoodcc committed:
fix for web server not associating the apache httpd media on creation

jmahmoodcc committed:
fix for web server not associating the apache httpd media on creation

jeddcuison committed:
Fix create and update web server tests

jeddcuison committed:
Apply write lock to distributeMedia and remove deprecated methods and properties

jeddcuison committed:
Remove duplicate mock definitions caused by conflict resolution

jeddcuison committed:
Fix failing test

jeddcuison committed:
Do not change the state

Details:

- Fixes bug wherein a new web server's state is changed to STOPPED when it is edited and then saved

jeddcuison committed:
Show MediaType display name instead of the enum name

jeddcuison committed:
Fix NPE

jeddcuison committed:
Replace apache httpd dir with the one from media that is stored in db

jeddcuison committed:
Construct httpd conf paths from values taken from the media table instead of from a property file

jeddcuison committed:
Fix failing tests

jeddcuison committed:
Make generated status path's protocol "http" and ping resource to "index.html"

jeddcuison committed:
Get port from db

jeddcuison committed:
Refactor code

- set busy icon outside of validation method since validation should only do validation
- use lastly to hide the busy icon and re-enable the dialog box
- wrap long line

jmahmoodcc committed:
Merge branch 'implement-tomcat-and-apache-webserver-media' of github.com:cerner/jwala into implement-tomcat-and-apache-webserver-media

jeddcuison committed:
Prevent users from doing something with the application until upload is complete

akinny commented on jwala-common/src/main/java/com/cerner/jwala/common/domain/model/webserver/WebServer.java:
```diff
@@ -121,7 +136,22 @@ public URI getStatusUri() {
+
+    private URI getDefaultUri() {
+        try {
+            return new URL("http://" + host + ":" + port + "/apache_pb.png").toURI();
```
Can you move the constants up near class declaration

jmahmoodcc committed:
fix web server not being associated with group after being created and remove deploy dir property

jmahmoodcc committed:
generate the web server and JVM with a media remote dir specified as a property, and update the postman tests to download the JDK media, Tomcat media, and Apache HTTPD media

jeddcuison committed:
Fix bug wherein file upload fails for IE

The reason why upload was failing is because IE gives the file's name with its full path which when concatenated with the repository path becomes an invalid path

jmahmoodcc committed:
fix slash missing after the remote.jwala.data.dir property, and remove some hard coded values

jmahmoodcc committed:
revert changes to allow templatizing the remote dir of media - don't allow templatized remote dir path, and do some clean up for associating the web servers with a group

jeddcuison committed:
Create the directory including any necessary but nonexistent parent directories

In addition, show the whole directory path when there is a failure in creating the directory to make troubleshooting easier

jeddcuison commented:
Please see #95
